### PR TITLE
fix(task-framework): update task result before canceling

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
If the task is immediately stopped, its resultJson field will not be updated. So before changing the task status, update the task results first.

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 1]
```diff
@@ -368,6 +368,13 @@ public boolean refreshLogMeta(Long id) {
         try {
             String executorEndpoint = executorEndpointManager.getExecutorEndpoint(je);
             DefaultTaskResult result = taskExecutorClient.getResult(executorEndpoint, JobIdentity.of(id));
+
+            // If the task is immediately stopped, its resultJson field would NOT be updated. So before changing
+            // the task status, update the task results first.
+            if (result != null) {
+                updateTaskResult(result, je);
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:375]: update result after logmeta saved

Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java]: done

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**